### PR TITLE
System publisher + SendEvent overloads

### LINKED ISSUE #17300: Add certificate support for partner teams
Partner teams will use client certs when authenticating with Event Grid. Since it is still required to pass in a credential, we will use "AzureSystemPublisher" to denote that the credentials should not be added to the request.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/17300

### CODE REVIEW

File: sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridKeyCredentialPolicy.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        public EventGridKeyCredentialPolicy(AzureKeyCredential credential, string name) : base(credential, name)
+        {
+            _credential = credential;
+            _name = name;
+        }
+
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            base.OnSendingRequest(message);
```
Comment (pakrym) [inline sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridKeyCredentialPolicy.cs:21]: This call would add the header.

File: sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridKeyCredentialPolicy.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Messaging.EventGrid
+{
+    internal class EventGridKeyCredentialPolicy : AzureKeyCredentialPolicy
```
Comment (pakrym) [inline sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridKeyCredentialPolicy.cs:8]: Do you really need to inherit from the AzureKeyCredentialPolicy?

Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridKeyCredentialPolicy.cs:8]: I guess not

**Review by pakrym — APPROVED** [review]

**Review by pakrym — APPROVED** [review]

**Pull request merged**